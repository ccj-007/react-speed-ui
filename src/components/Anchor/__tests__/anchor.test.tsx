import { render, fireEvent, screen } from '@testing-library/react';
import { Anchor } from '../../index';
import "@testing-library/jest-dom/extend-expect";
import React from 'react';

describe('test anchor component', () => {
  it('should render', () => {
    let jumpObj = [
      {
        queryId: "#header",
        title: "标题头部",
        key: 0,
      },
      {
        queryId: "#content",
        title: "内容区域",
        key: 1,
      },
      {
        queryId: "#content2",
        title: "内容2区域",
        key: 2,
      },
      {
        queryId: "#footer",
        title: "尾部区域",
        key: 3,
      },
    ];
    const { asFragment } = render(
      <>
        <Anchor jumpObj={jumpObj}></Anchor>
      </>
    );
    expect(asFragment()).toMatchSnapshot();
  })

  it('use speed-anchor className', () => { 
    const {container} = render(
        <Anchor />
    );
    expect(container.firstChild).toHaveClass('speed-anchor')
  });
})
