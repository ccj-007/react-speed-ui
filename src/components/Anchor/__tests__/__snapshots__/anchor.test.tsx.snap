// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test anchor component should render 1`] = `
<DocumentFragment>
  <div
    class="speed-anchor"
  >
    <div
      class="speed-anchor-line"
      style="height: 0px;"
    >
      <div
        class="speed-anchor-ballbox"
        style="top: 0px; height: 0px;"
      >
        <div
          class="speed-anchor-ball"
        />
      </div>
    </div>
    <div
      class="speed-anchor-content"
    >
      <div
        class="speed-anchor-item"
      >
        标题头部
      </div>
      <div
        class="speed-anchor-item"
      >
        内容区域
      </div>
      <div
        class="speed-anchor-item"
      >
        内容2区域
      </div>
      <div
        class="speed-anchor-item"
      >
        尾部区域
      </div>
    </div>
  </div>
</DocumentFragment>
`;
